### PR TITLE
Implement ignoreRedirects SW's new option

### PR DESCRIPTION
**Pull request by Tuch**
Hello,
In our project we're using `offline-plugin` and we really like it (Thank you!) but there is one issue with redirection.
For example when user is trying to open a page `/Page` we're redirecting him to `/page` but `offline-plugins` looks on it as on a cache case: https://github.com/NekR/offline-plugin/blob/master/src/misc/sw-template.js#L401

My propose is to use a new option `ignoreRedirects` to catch this case.

PS:
I've found one article with similar issue: 
https://medium.com/@boopathi/service-workers-gotchas-44bec65eab3f

### CODE REVIEW

**Comment by NekR** [comment]
Hi @Tuch! Thanks for your PR :-)

I think it even makes sense to make that default behavior because I cannot imagine any cases when someone would want a navigation fallback popping out instead of a redirect. Sounds like a bug in `offline-plugin`.

Unfortunately we cannot make it a default in v4 because that would be breaking/major change and v5 isn't even really for alpha release. I thinking now about adding this option only for v4 and remove it in v5. Hmm.. what do you think?

**Comment by Tuch** [comment]
@NekR sure, it's great! thank you.

**Comment by GGAlanSmithee** [comment]
@NekR @Tuch

Nice addition, I would agree that this probably should be considered to be the default behavior (for v5).  BTW, if this is only usable in conjunction with `navigateFallbackURL`, it might make more sense to allow for it to be an object instead?

```javascript
navigateFallbackURL: {
    ignoreRedirects: true
}
```

Less API surface, avoids conflicting config, but a bit more logic to implement.

**Comment by NekR** [comment]
> BTW, if this is only usable in conjunction with navigateFallbackURL, it might make more sense to allow for it to be an object instead?

Good catch, I think it should be done that way, yes 👍

**Comment by NekR** [comment]
Okay, `navigateFallbackURL` accepts a string so it should still accept it. But since this new option indeed affects only `navigateFallbackURL`, it should somehow be related to it. @GGAlanSmithee did you mean this?:
```js
navigateFallbackURL: {
    ignoreRedirects: true,
    url: '/'
}
```

I also think we should keep `ignoreRedirects` option for `navigateFallbackURL` in v5 but have it `true` by default, maybe someone would need to set it to `false`. Not sure about this though. Still thinking.

**Comment by GGAlanSmithee** [comment]
@NekR yes exactly. :+1:

**Comment by Tuch** [comment]
I would happy to any solution :) Let's choose something and i'll try to implement it.

**Comment by NekR** [comment]
@Tuch Okay, please rename the option to `navigateFallbackForRedirects` and make it `true` by default. Hence, to disable current behavior you will need to set `navigateFallbackForRedirects: false`. 

I thought about the name `navigateFallbackIgnoreRedirects`, but it isn't immediately clear how this option ignores redirects: 1) It ignores them and fallbacks to cache; 2) or it ignores them and doesn't fallback to cache.

@GGAlanSmithee I decided to keep that as a separate option, but name it more concrete. Having `navigateFallbackURL` as object adds unnecessary complexity (especially for a temporary option).


I hope that makes sense for both of you :-)

**Comment by Tuch** [comment]
@NekR i've changed the name and updated test snapshots

**Comment by NekR** [comment]
I'm not sure if wrote that clearly, but I meant that `navigateFallbackIgnoreRedirects` isn't obvious and we should name it `navigateFallbackForRedirects`.

**Comment by Tuch** [comment]
@NekR my fault, fixed :)

File: src/misc/sw-template.js [hunk 1]
```diff
@@ -398,7 +399,11 @@ function WebpackServiceWorker(params, helpers) {
     return fetching
       .catch(() => {})
       .then((response) => {
-        if (!response || !response.ok) {
+        const isOk = response && response.ok;
+        const isRedirect = response && response.type === 'opaqueredirect';
+        const useCache = !isOk && !(isRedirect && navigateFallbackForRedirects);
```
Comment (NekR) [inline src/misc/sw-template.js]: `const useCache = !isOk && (navigateFallbackForRedirects || !isRedirect);`

**Comment by NekR** [comment]
@Tuch I know I had to review everything at once, but it was hard to me until I saw clear picture with correct option name. Sorry for that.

**Comment by Tuch** [comment]
@NekR fixed, but i've changed a bit of logic:
```
  function handleNavigateFallback(fetching) {
    return fetching['catch'](function () {}).then(function (response) {
      var isOk = response && response.ok;
      var isRedirect = response && response.type === 'opaqueredirect';

      if (isOk || isRedirect && !navigateFallbackForRedirects) {
        return response;
      }

      if (DEBUG) {
        console.log('[SW]:', 'Loading navigation fallback [' + navigateFallbackURL + '] from cache');
      }

      return cachesMatch(navigateFallbackURL, CACHE_NAME);
    });
  }
```
- [x] by default `navigateFallbackForRedirects === true`
- [x] invert logic

please check it :)

**Comment by NekR** [comment]
Okay, I think it's fine now, thank you @Tuch! I'll make 4.7 release as soon as possible, but I have to fix https://github.com/NekR/offline-plugin/issues/231 first.

**Comment by Tuch** [comment]
@NekR, @GGAlanSmithee thank you!

**Comment by GGAlanSmithee** [comment]
@NekR understood, not worth the branching! Good work @Tuch :tada:

**Pull request merged**